tests/notify.rs

```diff
@@ -230,3 +230,31 @@ fn new_recommended() {
     Err(_) => assert!(false)
   }
 }
+
+// if this test builds, it means RecommendedWatcher is Send.
+#[test]
+fn test_watcher_send() {
+    let (tx, _) = channel();
+
+    let mut watcher: RecommendedWatcher = Watcher::new(tx).unwrap();
+
+    thread::spawn(move || {
+        watcher.watch(".").unwrap();
+    }).join().unwrap();
+}
+
+// if this test builds, it means RecommendedWatcher is Sync.
+#[test]
+fn test_watcher_sync() {
+    use std::sync::{ Arc, RwLock };
+
+    let (tx, _) = channel();
+
+    let watcher: RecommendedWatcher = Watcher::new(tx).unwrap();
+    let watcher = Arc::new(RwLock::new(watcher));
```

Comment:
I thought it was better to have a test that tried to do something useful than something clearly dummy. Maybe in the future we could also augment it to actually see if the watcher works. (tell me if you'd like me to do this now already)